### ISSUE TITLE
Offline CPUs

### ISSUE THREAD

**Issue by hyder365** [htop-dev/htop#853]
Hi. Would it be possible to add a configure flag (or even runtime option) to disable the showing of offline CPUs? It's taking up space in the terminal for no reason that I can see. Thanks.

**Comment by sergey-dryabzhinsky** [comment]
Do you refer to LXC/cgroups cpu limiting?

**Comment by hyder365** [comment]
No, this is about (for example) Intel CPUs with hyperthreading disabled. All the offline CPUs are still shown in htop, but wasting space in a small terminal.

**Comment by sergey-dryabzhinsky** [comment]
@hyder365 Did you tried lates code from master?
It already uses `/sys/devices/system/cpu/cpu*/online` lookups.

**Comment by hyder365** [comment]
Issue still present in 3.2.1.

**Comment by BenBE** [comment]
> Issue still present in 3.2.1.

Features and bug fixes included with a new release are usually listed as part of the change log, but normally also appear as part of the respective milestone. Have a look at https://github.com/htop-dev/htop/milestone/8 for the 3.2.1 release.

Currently this issue is not yet included with any upcoming milestone and given that this is a new feature there is no guaranteed schedule for it's implementation.

**Comment by hyder365** [comment]
I've built htop from git master as of today and still see the offline CPUs.
![g198kG6](https://user-images.githubusercontent.com/67167176/173195389-c5b72d58-b174-4c5f-b3fc-b458b62336f6.png)

**Comment by fasterit** [comment]
@hyder365: you will see that until **you** develop a patch, create a pull request, get that accepted and finally merged
(or somebody else thinks this is request is worth being implemented and does it)

**Comment by hyder365** [comment]
Ok. The comment by @sergey-dryabzhinsky led me to believe it was fixed in the master branch.

**Comment by alchemydc** [comment]
> @hyder365: you will see that until **you** develop a patch, create a pull request, get that accepted and finally merged (or somebody else thinks this is request is worth being implemented and does it)

I think it's worth doing also, fwiw.  As will likely any other heavy user of LXC.

**Comment by LukeWheeldon** [comment]
I am impacted as well, having just upgraded to Debian Bookworm in an environment where virtual cores are disabled.

<img width="1136" alt="Screenshot 2023-10-14 at 16 25 51" src="https://github.com/htop-dev/htop/assets/61089799/0cdb55b1-4a73-4004-b620-c4862e0cd73f">

**Comment by LukeWheeldon** [comment]
In case it can help anyone, there's a way around this issue.

First, you need to configure htop and remove all the CPU displayed by default, and re-add them one by one, half on the left, half on the right.

Then, you need to save the configuration so you don't have to repeat the operation (it takes a little while). On MacOS/iTerm I could do that with the keyboard combination "fn+F10", other systems likely have similar ways of saving the configuration to disk.

<img width="1305" alt="Screenshot 2023-10-19 at 20 33 49" src="https://github.com/htop-dev/htop/assets/61089799/14a5718d-7c61-4276-8b0a-57b2f4fc89df">

**Comment by moralground** [comment]
Unfortunate that such workarounds are needed for so basic a feature after two years.

**Comment by BenBE** [comment]
> Unfortunate that such workarounds are needed for so basic a feature after two years.

As this is a new feature its usually not high on the list of priorities, unless people supply a patch with it. Also with any feature involving additional items of configuration we are usually quite reluctant, as this adds a maintenance cost going forward. And the last item is the automatic height adjustment required for this feature to be "dynamically" hiding the meters is a bit more involved than might be visible on first glance: You not only have to determine the number of visible meters dynamically (doable), calculate the needed height from this (doable), but also map each meter to its proper spot (non-trivial if truly dynamic).

If you want to speed up the inclusion of this feature a working patch is a good first step in that direction.

**Comment by sln-hubert** [comment]
Bump, please.
